segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLogFactory.java

```diff
@@ -98,13 +99,26 @@ public void initialize() throws DurableDataLogException {
     }
 
     @Override
-    public DurableDataLog createDurableDataLog(int containerId) {
+    public DurableDataLog createDurableDataLog(int logId) {
         Preconditions.checkState(this.bookKeeper.get() != null, "BookKeeperLogFactory is not initialized.");
-        return new BookKeeperLog(containerId, this.zkClient, this.bookKeeper.get(), this.config, this.executor);
+        return new BookKeeperLog(logId, this.zkClient, this.bookKeeper.get(), this.config, this.executor);
+    }
+
+    /**
+     * Creates a new DebugLogWrapper that can be used for debugging purposes. This should not be used for regular operations.
+     *
+     * @param logId Id of the Log to create a wrapper for.
+     * @return A new instance of the DebugLogWrapper class.
+     */
+    @VisibleForDebugging
+    public DebugLogWrapper createDebugLogWrapper(int logId) {
+        Preconditions.checkState(this.bookKeeper.get() != null, "BookKeeperLogFactory is not initialized.");
+        return new DebugLogWrapper(logId, this.zkClient, this.bookKeeper.get(), this.config, this.executor);
     }
 
     @VisibleForTesting
-    BookKeeper getBookKeeperClient() {
+    @VisibleForDebugging
```

Comment:
I think this should replace VisableForTesting here.

Reply:
Same reasoning as above.